cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/query_enhancements/dataset_selector.spec.js

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DATASOURCE_NAME, START_TIME, END_TIME } from '../../../../../utils/apps/constants';
+import { SECONDARY_ENGINE } from '../../../../../utils/constants';
+import { getRandomizedWorkspaceName } from '../../../../../utils/apps/query_enhancements/shared';
+
+const workspace = getRandomizedWorkspaceName();
+
+describe('dataset selector', { scrollBehavior: false }, () => {
+  before(() => {
+    cy.setupTestData(
+      SECONDARY_ENGINE.url,
+      ['cypress/fixtures/query_enhancements/data-logs-1/data_logs_small_time_1.mapping.json'],
+      ['cypress/fixtures/query_enhancements/data-logs-1/data_logs_small_time_1.data.ndjson']
+    );
+
+    // Add data source
+    cy.addDataSource({
+      name: `${DATASOURCE_NAME}`,
+      url: `${SECONDARY_ENGINE.url}`,
+      authType: 'no_auth',
+    });
+  });
+  after(() => {
+    cy.deleteDataSourceByName(`${DATASOURCE_NAME}`);
+    cy.deleteIndex('data_logs_small_time_1');
+  });
+  beforeEach(() => {
+    // Create workspace
+    cy.deleteWorkspaceByName(`${workspace}`);
+    cy.visit('/app/home');
+    cy.createInitialWorkspaceWithDataSource(`${DATASOURCE_NAME}`, `${workspace}`);
+    cy.navigateToWorkSpaceSpecificPage({
+      workspaceName: workspace,
+      page: 'discover',
+      isEnhancement: true,
+    });
+  });
+
+  afterEach(() => {
+    cy.deleteWorkspaceByName(`${workspace}`);
+  });
+
+  // TODO: There is an issue selecting indexes in 2.x, but it works on main CI/CD
+  describe.skip('select indices', () => {
```

Comment:
added a skip here due to issues with selecting index as dataset

Reply:
Is it due to some un-backported PR? we could backport that PR first then enable these tests

Reply:
@abbyhu2000 's backport fixed this issue for me. Removing the skips next commit